packages/cojson-storage-sqlite/src/tests/sqlite.test.ts

```diff
@@ -0,0 +1,321 @@
+import { SQLiteNode } from "../index.js";
+import { toSimplifiedMessages } from "./messagesTestUtils.js";
+import { trackMessages } from "./testUtils.js";
+
+const Crypto = await WasmCrypto.create();
+
+async function createSQLiteStorage(defaultDbPath?: string) {
+  const dbPath = defaultDbPath ?? join(tmpdir(), `test-${randomUUID()}.db`);
+
+  if (!defaultDbPath) {
+    onTestFinished(() => {
+      unlinkSync(dbPath);
+    });
+  }
+
+  return {
+    peer: await SQLiteNode.asPeer({
+      filename: dbPath,
+    }),
+    dbPath,
+  };
+}
+
+test("Should be able to initialize and load from empty DB", async () => {
+  const agentSecret = Crypto.newRandomAgentSecret();
+
+  const node = new LocalNode(
+    new ControlledAgent(agentSecret, Crypto),
+    Crypto.newRandomSessionID(Crypto.getAgentID(agentSecret)),
+    Crypto,
+  );
+
+  node.syncManager.addPeer((await createSQLiteStorage()).peer);
+
+  await new Promise((resolve) => setTimeout(resolve, 200));
+
+  expect(node.syncManager.peers.storage).toBeDefined();
+});
+
+test("should sync and load data from storage", async () => {
+  const agentSecret = Crypto.newRandomAgentSecret();
+
+  const node1 = new LocalNode(
+    new ControlledAgent(agentSecret, Crypto),
+    Crypto.newRandomSessionID(Crypto.getAgentID(agentSecret)),
+    Crypto,
+  );
+
+  const node1Sync = trackMessages(node1);
+
+  const { peer, dbPath } = await createSQLiteStorage();
+
+  node1.syncManager.addPeer(peer);
+
+  const group = node1.createGroup();
+
+  const map = group.createMap();
+
+  map.set("hello", "world");
+
+  await new Promise((resolve) => setTimeout(resolve, 200));
+
+  expect(
+    toSimplifiedMessages(
+      {
+        Map: map.core,
+        Group: group.core,
+      },
+      node1Sync.messages,
+    ),
+  ).toMatchInlineSnapshot(`
+    [
+      "client -> CONTENT Group header: true new: After: 0 New: 3",
+      "client -> CONTENT Map header: true new: After: 0 New: 1",
+    ]
+  `);
+
+  node1Sync.restore();
+
+  const node2 = new LocalNode(
+    new ControlledAgent(agentSecret, Crypto),
+    Crypto.newRandomSessionID(Crypto.getAgentID(agentSecret)),
+    Crypto,
+  );
+
+  const node2Sync = trackMessages(node2);
+
+  const { peer: peer2 } = await createSQLiteStorage(dbPath);
+
+  node2.syncManager.addPeer(peer2);
+
+  const map2 = await node2.load(map.id);
+  if (map2 === "unavailable") {
+    throw new Error("Map is unavailable");
+  }
+
+  expect(map2.get("hello")).toBe("world");
+
+  expect(
+    toSimplifiedMessages(
+      {
+        Map: map.core,
+        Group: group.core,
+      },
+      node2Sync.messages,
+    ),
+  ).toMatchInlineSnapshot(`
+    [
+      "client -> LOAD Map sessions: empty",
+      "storage -> KNOWN Group sessions: header/3",
+      "storage -> CONTENT Group header: true new: After: 0 New: 3",
+      "client -> KNOWN Group sessions: header/3",
+      "storage -> KNOWN Map sessions: header/1",
+      "storage -> CONTENT Map header: true new: After: 0 New: 1",
```

Comment:
We can probably optimize away those `storage -> KNOWN` messages because the content is enough to update the knownState